xarray/tests/test_dataset.py

```diff
@@ -6217,6 +6217,13 @@ def test_polyfit_output(self) -> None:
         out = ds.polyfit("time", 2)
         assert len(out.data_vars) == 0
 
+    def test_polyfit_weighted(self) -> None:
+        ds = create_test_data(seed=1)
+        ds_copy = ds.copy(deep=True)
+
+        ds.polyfit("dim2", 2, w=np.arange(ds.sizes["dim2"]))
+        xr.testing.assert_identical(ds, ds_copy)
```

Comment:
Maybe add the issue as comment to justify why we check this .

Reply:
Good point, done.
My original plan was to add that check in pre-existing polyfit tests, but it looks like polyfit with weights is not tested elswehere.